src/dump.c

```diff
@@ -2588,7 +2591,8 @@ static jl_method_t *jl_lookup_method_worldset(jl_methtable_t *mt, jl_datatype_t
     jl_method_t *_new;
     while (1) {
         _new = (jl_method_t*)jl_methtable_lookup(mt, sig, world);
-        assert(_new && jl_is_method(_new));
+        if (!(_new && jl_is_method(_new)))
+            return NULL; // the method wasn't found, probably because it was disabled
```

Comment:
This function absolutely must return a valid object, otherwise we're leaving a field somewhere with uninitialized memory (also, incidentally, with the wrong typeof).

Reply:
I do check for a NULL return in all callers of this method, is that not good enough? The overall strategy I took is to fail in a way that signals to `jl_insert_backedges` and have it drop any dependent methods.

If it really must return a valid method, then I could return to an intermediate version of this PR (before I discovered what I thought was a clever strategy). There I added a `max_world_mask` to `jl_typemap_assoc/lookup_by_type` and then re-ran the lookup to ignore `max_world` in deciding whether a method fits. Not quite sure what it will grab, but it will grab something.